### DISCUSSION TITLE
The future of Micropython and Pyboard

### DISCUSSION THREAD

**Discussion by micGHkent** [micropython/micropython#11589] [STM32 / Pyboard]
I am very new to 32 bit Micros and the available hardware and software.  I have just searched worldwide for a Pyboard and only found 'not in stock's'.  Also there appears to be some new versions of Python popping up.  I am trying to avoid a rabbit hole to nowhere.
Comments please.  
Mick  (old retired EE)

**Comment by peterhinch** [comment]
Pyboards are like hens' teeth at the moment because of STM chip availability issues. MicroPython supports numerous platforms, including the various ESP32 boards. I would look at the [Raspberry Pico and Pico W](https://www.raspberrypi.com/products/raspberry-pi-pico/): if you're an EE like myself you'll like the capability for hard ISR's and also their unique IO FSM's. Not to mention their super low prices...

**Reply by micGHkent** [reply]
Hello Peter,  Thanks for your response.  Like I said I am very new to the 32 bit field.  I was a machine controls engineer so a lot of ME mixed in with the EE.  My hobby choice of controllers was the Microchip PIC's and I programmed with a compiler from CCS and Microchips own, both in 'C'.  I have spent a good bit of time on the ST website and bought a new board with a new series of the STM32, the H503.  The Nucleo-H503RB was just announced this month and I already have one on my bench.  ST has a free IDE and programmer as well as established (read expensive) third partys, IAR and Arm Keil.  They all work in C and C++ and don't mention Python or Micropython.  They probably don't like 'Open source' due to the lack of a business organization.  Anyway both Adafruit and Sparkfun have plenty of STM32 modules in stock.
Thank-you for the Raspberry Pico suggestion.  I'll take a good look at that choice.  I do think I have a Raspberry Pi in my collection of old projects.  I see a Raspberry Pi camera on my shelf.  I want a wireless connection due to much trouble with Win 11 com ports.       Mick

**Reply by peterhinch** [reply]
Raspberry Pi and Pico are entirely different. As you doubtless know, the Pi runs Linux whereas the Pico is a "bare metal" device with no OS. The Pico can therefore offer fast realtime performance. The Pico W has WiFi, provided by a separate chip. It therefore retains its "bare metal" credentials. (The Pyboard D uses a similar design, of a separate WiFi chip).

At risk of being contentious the solution to Win 11 com ports is simple. Run Linux. :)

**Comment by jimmo** [comment]
The pyboard product and the MicroPython project run mostly independently from eachother. MicroPython runs on hundreds of other boards across several chip vendors and families. See https://micropython.org/download

The pyboard is a reference platform of sorts, as well as being a revenue source to help fund the development of MicroPython. As Peter said, the chip shortage made it very hard to get the STM32F4 and STM32F7 parts as well as a few other support parts. There is a new pyboard in the works based on the super-low-power Ambiq Apollo chips (that was hinted at in the 10 year poster: https://micropython.org/resources/MicroPython10YearsPoster.pdf), and we hope to resume distribution of the original ST pyboards at some stage soon.

>  I have spent a good bit of time on the ST website and bought a new board with a new series of the STM32, the H503. The Nucleo-H503RB was just announced this month and I already have one on my bench.

Unfortunately we haven't quite got support for the H5 ready yet (like you say, it's very new!). MicroPython support is in the works -- see https://github.com/micropython/micropython/pull/11138
So at some point quite soon you'll see the STM32H573I_DK firmware show up on the downloads page, and it should be relatively straightforward to add support for the H5 Nucleos after that.

>  I do think I have a Raspberry Pi in my collection of old projects. I see a Raspberry Pi camera on my shelf. 

Just in case, it sounds like you might be talking about the regular Raspberry Pi (i.e. not the Pico). MicroPython doesn't support this (we only support the Pico), but CircuitPython does have full Raspberry Pi support, so might be worth checking that out.

**Reply by ruffrecords** [reply]
Sorry to revive such an old thread but it seems very pertinent to my question. I purchased a pyboard in the original kickstarter but never did much more than tinker with it. Since then I have tinkered with the Pico and ESP32 versions of micropython. However, now I have a real project that would be ideal for micropython but it does require a decent DAC and the pyboard seems to be the only one I can find with a 12 bit DAC on board. However, all the pyboards on the official web store are listed as out of stock. Yes, there are loads of Chinese clones available on eBay but I would much rather buy British and support the project. What is the situation regarding the availability of official pyboards?

**Reply by robert-hh** [reply]
Boards by WeACT are usually of good quality. And besides Pyboard clones, there are other boards based on ST32F405.
And there is for example the SAMD51 port, which offers a 12 Bit DAC with 2 channels.

**Reply by ruffrecords** [reply]
Thank you for the information. I have ordered a couple of WeAct boards.

**Reply by ruffrecords** [reply]
The  WeACT boards have arrived but I just cannot get them to connect to Thonny. I cannot even persuade it to appear as a USB drive. Is there something special I need to do?

Cheers

Ian

**Reply by robert-hh** [reply]
Which board did you order and how did you install MicroPython?

**Reply by ruffrecords** [reply]
I ordered the WeAct Studio STM32F405RGT6 Development Board. I ASSumed it came with MicroPython installed. It appears to have a blinky program running because the blue LED blinks regularly. If I were to (re)load Micropython on it then I would use dfu-util to do it. I have downloaded PYBV10-20251209-v1.27.0.dfu. Do they normally come unprogrammed?

**Reply by peterhinch** [reply]
I don't know that particular board but in general you need to install MicroPython using the firmware and installation instructions on the [MicroPython website](https://micropython.org/download/).

**Reply by robert-hh** [reply]
Push & Hold the user button, press reset and then holding the user button for a while should set the board in firmware upload mode.  You can then try with `dfu-util -l` if the board shows up.

**Reply by scruss** [reply]
> WeAct Studio STM32F405RGT6 Development Board

This doesn't have a stock board definition. WeAct's repo [WeActStudio.STM32F4_64Pin_CoreBoard](https://github.com/WeActStudio/WeActStudio.STM32F4_64Pin_CoreBoard/tree/master/Examples/STM32F405/mpy) has an old-ish MicroPython hex file for it, along with a board definition that may or may not still work.

(might be best to start a new thread about it, since this isn't about Pyboards)

**Reply by robert-hh** [reply]
I have ordered the STM32F405RGT6 board as well and assemble a preliminary dfu file for it. It boots, shows REPL and pops up as drive. Attached is the dfu file and the actual state of board files. To enable dfu mode, push and hold the B0 switch (the middle) and push the NR button short. Then release all buttons. `dfu-util -l` should then show a list of devices.  Then upload the firmware with: `dfu-util --alt 0 -D firmware.dfu`. Push reset after the upload is finished.

[WEACT_F405.zip](https://github.com/user-attachments/files/24695042/WEACT_F405.zip)

Edit: Updated.

**Comment by Oliver2003Taylor** [comment]
The original Pyboard has become hard to find simply because the ecosystem has moved on a bit, not because it’s obsolete. most people today skip the Pyboard and start with something that’s easy to source and well supported. The Raspberry Pi Pico (or Pico W) is very popular and straightforward, and ESP32 boards are everywhere if you want more features. if you prefer to stay with STM32, Nucleo boards still work fine with MicroPython, just with a bit more setup.

As for the “new Python versions,” most of those are just board-specific forks. Sticking to the official MicroPython builds from micropython.org will keep things simple and stable.

In short: pick commonly available hardware, use official MicroPython, and you’ll avoid most of the dead ends.
